go-controller/cmd/ovn-kube-util/app/readiness-probe.go

```diff
@@ -103,15 +103,19 @@ func ovnSBDBReadiness(target string) error {
 }
 
 func ovnNorthdReadiness(target string) error {
+	// checking version works as it conntects to northd and returns the version
+	// if northd isn't ready, version may fail to return
 	_, _, err := util.RunOVNAppctlWithTimeout(5, "-t", target, "version")
 	if err != nil {
 		return fmt.Errorf("failed to get version from %s: (%v)", target, err)
 	}
 	return nil
 }
 
-func ovnNbCtldReadiness(target string) error {
-	_, _, err := util.RunOVNAppctlWithTimeout(5, "-t", "ovn-nbctl", "version")
+func ovnNbCtlReadiness(target string) error {
+	// checking version works as it conntects to nbctld and returns the version
```

Comment:
"connects to the ovn-nbctl daemon and returns"

Reply:
Right, it definitely confused me initially as well. I had to read up about it. I have added a note. PTAL. :)